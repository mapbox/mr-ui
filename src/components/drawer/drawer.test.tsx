import React from 'react';
import Drawer from './drawer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

describe('Drawer', () => {
  const user = userEvent.setup();
  describe('basic', () => {
    const mockedOnExit = jest.fn();
    const props = {
      onExit: mockedOnExit,
      children: <div>Drawer body</div>
    } as const;

    test('renders', () => {
      const { baseElement } = render(<Drawer {...props} />);
      expect(baseElement).toMatchSnapshot();
    });

    test('fires onExit', async () => {
      render(<Drawer {...props} />);
      await user.click(screen.getByTestId('drawer-overlay'));
      expect(mockedOnExit).toHaveBeenCalledTimes(1);
    });
  });

  describe('basic, left', () => {
    const mockedOnExit = jest.fn();
    const props = {
      onExit: mockedOnExit,
      children: <div>Drawer body</div>,
      side: 'left'
    } as const;

    test('renders', () => {
      const { baseElement } = render(<Drawer {...props} />);
      expect(baseElement).toMatchSnapshot();
    });

    test('fires onExit', async () => {
      render(<Drawer {...props} />);
      await user.click(screen.getByTestId('drawer-overlay'));
      expect(mockedOnExit).toHaveBeenCalledTimes(1);
    });
  });

  describe('basic, right', () => {
    const mockedOnExit = jest.fn();
    const props = {
      onExit: mockedOnExit,
      children: <div>Drawer body</div>,
      side: 'right'
    } as const;

    test('renders', () => {
      const { baseElement } = render(<Drawer {...props} />);
      expect(baseElement).toMatchSnapshot();
    });

    test('fires onExit', async () => {
      render(<Drawer {...props} />);
      await user.click(screen.getByTestId('drawer-overlay'));
      expect(mockedOnExit).toHaveBeenCalledTimes(1);
    });
  });
});