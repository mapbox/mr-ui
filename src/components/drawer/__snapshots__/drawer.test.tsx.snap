// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer basic renders 1`] = `
<body
  data-scroll-locked="1"
  style="pointer-events: none;"
>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <div
      data-testid="event-trap"
    />
  </div>
  <div
    class="fixed top bottom left right bg-darken50 "
    data-state="open"
    data-testid="drawer-overlay"
    style="pointer-events: auto; z-index: auto;"
  >
    <div
      aria-describedby="radix-:r2:"
      aria-labelledby="radix-:r1:"
      class="overflow-auto scroll-styled fixed top bottom  left round-tr round-br bg-white"
      data-state="open"
      id="radix-:r0:"
      role="dialog"
      style="pointer-events: auto;"
      tabindex="-1"
    >
      <div>
        Drawer body
      </div>
    </div>
  </div>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
</body>
`;

exports[`Drawer basic, left renders 1`] = `
<body
  data-scroll-locked="1"
  style="pointer-events: none;"
>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <div
      data-testid="event-trap"
    />
  </div>
  <div
    class="fixed top bottom left right bg-darken50 "
    data-state="open"
    data-testid="drawer-overlay"
    style="pointer-events: auto; z-index: auto;"
  >
    <div
      aria-describedby="radix-:r8:"
      aria-labelledby="radix-:r7:"
      class="overflow-auto scroll-styled fixed top bottom  left round-tr round-br bg-white"
      data-state="open"
      id="radix-:r6:"
      role="dialog"
      style="pointer-events: auto;"
      tabindex="-1"
    >
      <div>
        Drawer body
      </div>
    </div>
  </div>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
</body>
`;

exports[`Drawer basic, right renders 1`] = `
<body
  data-scroll-locked="1"
  style="pointer-events: none;"
>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
  <div
    aria-hidden="true"
    data-aria-hidden="true"
  >
    <div
      data-testid="event-trap"
    />
  </div>
  <div
    class="fixed top bottom left right bg-darken50 "
    data-state="open"
    data-testid="drawer-overlay"
    style="pointer-events: auto; z-index: auto;"
  >
    <div
      aria-describedby="radix-:re:"
      aria-labelledby="radix-:rd:"
      class="overflow-auto scroll-styled fixed top bottom  right round-tl round-bl bg-white"
      data-state="open"
      id="radix-:rc:"
      role="dialog"
      style="pointer-events: auto;"
      tabindex="-1"
    >
      <div>
        Drawer body
      </div>
    </div>
  </div>
  <span
    aria-hidden="true"
    data-aria-hidden="true"
    data-radix-focus-guard=""
    style="outline: none; opacity: 0; position: fixed; pointer-events: none;"
    tabindex="0"
  />
</body>
`;
